pinot-core/src/main/java/org/apache/pinot/core/operator/docidsets/OrDocIdSet.java

```diff
@@ -77,6 +77,7 @@ public BlockDocIdIterator iterator() {
         sortedDocIdIterators.add((SortedDocIdIterator) docIdIterator);
         numEntriesScannedForNonScanBasedDocIdSets += docIdSet.getNumEntriesScannedInFilter();
       } else if (docIdIterator instanceof BitmapBasedDocIdIterator) {
+        bitmapBasedDocIdIterators.add((BitmapBasedDocIdIterator) docIdIterator);
```

Comment:
@Jackie-Jiang : fyi: this was a regression in 1.2. For one of our use-cases this increased the latency by almost 100x. The issue is that this can defer the evaluation of the filters in OR, and if the parent of the Or Filter is an And with one side of it being a heavy scan, then the scan would end up running before the Or is run.